src/main/fox_hud.c

```diff
@@ -4812,10 +5642,572 @@ void func_80095538(Actor* actor, s32 arg1) {
     actor->unk_0B6 = 47;
     Object_SetInfo(&actor->info, actor->obj.id);
 }
-#else
-#pragma GLOBAL_ASM("asm/us/nonmatchings/main/sf_hud/func_80095538.s")
-#endif
+void func_80095604(Player* player) {
+    s32 i;
+    s32 j;
+    Actor* actor;
+    f32 x;
+    f32 y;
+    f32 z;
+    f32 x1;
+    f32 y1;
+    f32 z1;
+    s32 pad;
+    Vec3f src;
+    Vec3f dest;
+    s32 shield;
+
+    actor = &gActors[0];
+
+    switch (player->unk_1D0) {
+        case 0:
+            D_80177A80 = 0;
+            player->unk_280 = 0;
+            player->unk_1D0 = 1;
+
+            D_80177A48[1] = gBosses[0].obj.pos.x;
+            D_80177A48[2] = gBosses[0].obj.pos.y;
+            D_80177A48[3] = gBosses[0].obj.pos.z;
+            D_80177A48[4] = 0.0f;
+            D_80177A48[5] = 0.0f;
+            D_80177A48[6] = 0.05f;
+
+            D_80177A48[7] = 0.0f;
+            D_80177A48[8] = 0.0f;
+            D_80177A48[9] = 0.0f;
+            D_80177A48[0] = 0.0f;
+
+        case 1:
+            if (D_80177A80 > 240) {
+                D_80178348 = D_80178350 = D_80178354 = 255;
+                D_80178358 = 255;
+                D_8017835C = 16;
+                if (D_80177A80 == 280) {
+                    player->unk_1D0 = 10;
+                }
+            }
+
+            if ((D_80177A80 >= 30) && (D_80177A80 < 90)) {
+                D_80177A48[5] -= 0.6f;
+                if ((D_80177A48[5]) < 0.0f) {
+                    D_80177A48[5] += 360.0f;
+                }
+                if ((D_80177A48[5]) > 360.0f) {
+                    D_80177A48[5] -= 360.0f;
+                }
+                src.x = 1000.0f;
+                src.y = 300.0f;
+                src.z = 1000.0f;
+
+                Matrix_Translate(gCalcMatrix, D_80177A48[1], D_80177A48[2], D_80177A48[3], 0);
+                Matrix_RotateY(gCalcMatrix, -(M_DTOR * D_80177A48[5]), 1);
+                Matrix_MultVec3f(gCalcMatrix, &src, &dest);
+            }
+
+            if (D_80177A80 < 30) {
+                dest.x = gBosses[0].obj.pos.x;
+                dest.y = player->pos.y;
+                dest.z = gBosses[0].obj.pos.z + 1000.0f;
+            }
+            if (D_80177A80 >= 90) {
+                dest.x = player->camEye.x;
+                dest.y = player->camEye.y;
+                dest.z = player->camEye.z + 200.0f - D_80177D20;
+            }
+
+            Math_SmoothStepToAngle(&player->unk_130, 0.0f, 0.05f, 1.2f, 0.0001f);
+            Math_SmoothStepToAngle(&player->unk_12C, 0.0f, 0.05f, 1.2f, 0.0001f);
+            Math_SmoothStepToF(&player->unk_034, 0.0f, 0.1f, 3.0f, 0.0f);
+            Math_SmoothStepToAngle(&player->unk_4D8, 0.0f, 0.1f, 20.0f, 0.0f);
+
+            x = player->pos.x - dest.x;
+            y = player->pos.y - dest.y;
+            z = player->pos.z - dest.z;
+
+            y1 = Math_RadToDeg(Math_Atan2F(x, z));
+            z = sqrtf(x * x + z * z);
```

Comment:
SQ macro. Also this temp reuse ended up being required, right?

Reply:
yes